add two more methods to the monotonic section

bradleypeabody opened the pull request:
After reading through everything again, while the monotonic/counter section works as it stands now, I think it would greatly benefit from two other specific suggested methods for achieving monotonicity.

The first I think is beneficial because it is trivial to implement (just keep generating random values until you get one that is greater than the last), and implementors prioritizing simplicity would benefit from it.

And the second brings back a concept that I originally proposed in the initial draft for UUID v7 (using bits in the UUID that immediately follow the timestamp to encode more timestamp precision in a way that is time ordered), but later removed because it was often misunderstood.  However, I think the key benefit to this approach (just using the additional precision available in the system clock for what it is meant for) still exists and makes sense, and having it here as an optional feature so implementations are not required to understand and use it if they don't need to - I believe that tradeoff makes a lot of sense.  I tried to keep it super simple by just giving an example of how to use it, and people can infer the underlying math/theory or not as needed.

bradleypeabody commented on draft-ietf-uuidrev-rfc4122bis.md:
```diff
@@ -1219,7 +1219,8 @@ before creating a new UUID. Distributed nodes are discussed in
 Implementations SHOULD choose one method for single-node UUID implementations
-that require batch UUID creation.
+that require batch UUID creation, or are otherwise concerned about monotonicity
+with high frequency UUID generation.
```
This is the same edit from here https://github.com/ietf-wg-uuidrev/rfc4122bis/pull/79#discussion_r1161177270

jimfenton commented on draft-ietf-uuidrev-rfc4122bis.md:
```diff
@@ -1251,6 +1252,42 @@ Monotonic Random (Method 2):
+  of multiple UUID values using the same clock tick, simply regenerate
+  a new random value as many times as needed in order to produce a UUID that is
+  ordered after the last one.  The primary benefit of this approach is simplicity
+  of implementation. The down sides to this approach include the fact that
```
```suggestion
  of implementation. One downside to this approach is the fact that
```

jimfenton commented on draft-ietf-uuidrev-rfc4122bis.md:
```diff
@@ -1251,6 +1252,42 @@ Monotonic Random (Method 2):
+  a new random value as many times as needed in order to produce a UUID that is
+  ordered after the last one.  The primary benefit of this approach is simplicity
+  of implementation. The down sides to this approach include the fact that
+  generation will become increasing slow for a given timestamp value as
```
```suggestion
  generation will become increasingly slow for a given timestamp value as
```

jimfenton commented on draft-ietf-uuidrev-rfc4122bis.md:
```diff
@@ -1251,6 +1252,42 @@ Monotonic Random (Method 2):
+  ordered after the last one.  The primary benefit of this approach is simplicity
+  of implementation. The down sides to this approach include the fact that
+  generation will become increasing slow for a given timestamp value as
+  more attempts are required to product monotonic output, and the fact that
```
```suggestion
  more attempts are required to produce monotonic output, and the fact that
```

jimfenton reviewed (changes requested):
Mostly editorial fixes suggested; I think we need more input on this. They both need to be expressed in normative language

Method 4 mixes the normative requirements with an example. Suggest pulling the example into one or more separate paragraphs so the reader can distinguish between what they are required to do and the example of how to do it.

bradleypeabody commented:
@jimfenton Thanks for the feedback, I did the edits you suggested and split up Method 4 into a few paragraphs and added a normative description as you suggest.  Let me know if there's anything else you think I should adjust immediately and otherwise I agree it would be good to get more feedback from others and go from there.

kyzer-davis commented:
Verified Method 3 and 4 render properly with author-tools in HTML/TXT formats. Merging down.
